user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -282,9 +348,49 @@ private void addEventListeners() {
           displayRegistrationSuccess();
         });
       });
-      registerButton.addClickListener(event -> batchRegistrationDialog.open());
-      batchRegistrationDialog.addCancelEventListener(
-          event -> batchRegistrationDialog.close());
+      registerButton.addClickListener(SampleDetailsComponent.this::onRegisterButtonClicked);
+    }
+
+
+    private Result<?, ?> registerBatchAndSamples(BatchRegistrationContent batchRegistrationContent,
+        List<SampleRegistrationContent> sampleRegistrationContent) {
+      return registerBatchInformation(batchRegistrationContent).onValue(
+          batchId -> {
+            List<SampleRegistrationRequest> sampleRegistrationsRequests = createSampleRegistrationRequests(
+                batchId, batchRegistrationContent.experimentId(), sampleRegistrationContent);
+            registerSamples(sampleRegistrationsRequests);
+          });
+    }
+
+    private Result<BatchId, ResponseCode> registerBatchInformation(
+        BatchRegistrationContent batchRegistrationContent) {
+      return batchRegistrationService.registerBatch(batchRegistrationContent.batchLabel(),
+              batchRegistrationContent.isPilot(), context.projectId().orElseThrow())
+          .onError(responseCode -> displayRegistrationFailure())
+          .onValue(batchId -> displayRegistrationSuccess());
+    }
```

Comment:
Either use these methods in the onRegisterDialogConfirmed method or remove them 🗑️

Reply:
They are now removed :)